src/app/login/login.service.ts

```diff
@@ -32,8 +32,12 @@ export class LoginService {
           observable.next(user);
           observable.complete();
         },
-        (error: HttpErrorResponse) => {
-          this.alertService.detailedError(error);
+        (error) => {
```

Comment:
It would be nice to preserve the type, now it's being treated as `any` by the compiler: `(error: HttpErrorResponse | undefined)`.

Reply:
The cause of the bug is that the `error` object is not necessarily an `HttpErrorResponse`, it can be some other type of object, for example the empty object `{}` that is produced when the user closes the login window, essentially whatever the underlying `ng2-ui-auth` decides to return.  So, IMHO, it's appropriate to not type the argument.